src/legacy/core_plugins/input_control_vis/public/components/vis/range_control.js

```diff
@@ -17,175 +17,93 @@
-    {
-      min: control.min,
-      max: control.min
-    };
-  const state = {
-    sliderValue,
-    minValue: control.hasValue() ? control.value.min : '',
-    maxValue: control.hasValue() ? control.value.max : '',
-    isRangeValid: true,
-    errorMessage: '',
-  };
-  return state;
-};
 
-class RangeControlUi extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = toState(props);
+function roundWithPrecision(value, decimalPlaces, roundFunction) {
+  if (decimalPlaces <= 0) {
+    return roundFunction(value);
   }
 
-  componentWillReceiveProps = (nextProps) => {
-    this.setState(toState(nextProps));
-  };
-
-  handleOnChange = (value) => {
-    this.setState({
-      sliderValue: value,
-      minValue: value.min,
-      isRangeValid: true,
-      maxValue: value.max,
-      errorMessage: '',
-    });
-  };
-
-  handleOnChangeComplete = (value) => {
-    this.props.stageFilter(this.props.controlIndex, value);
-  };
-
-  handleMinChange = (evt) => {
-    this.handleChange(parseFloat(evt.target.value), this.state.maxValue);
-  };
-
-  handleMaxChange = (evt) => {
-    this.handleChange(this.state.minValue, parseFloat(evt.target.value));
-  };
-
-  handleChange = (min, max) => {
-    min = isNaN(min) ? '' : min;
-    max = isNaN(max) ? '' : max;
-
-    const isMinValid = min !== '';
-    const isMaxValid = max !== '';
-    let isRangeValid = true;
-    let errorMessage = '';
-
-    if ((!isMinValid && isMaxValid) || (isMinValid && !isMaxValid)) {
-      isRangeValid = false;
-      errorMessage = this.props.intl.formatMessage({
-        id: 'inputControl.vis.rangeControl.minMaxValidErrorMessage',
-        defaultMessage: 'both min and max must be set'
-      });
-    }
+  let results = value;
+  results = results * Math.pow(10, decimalPlaces);
+  results = roundFunction(results);
+  results = results / Math.pow(10, decimalPlaces);
+  return results;
+}
 
-    if (isMinValid && isMaxValid && max < min) {
-      isRangeValid = false;
-      errorMessage = this.props.intl.formatMessage({
-        id: 'inputControl.vis.rangeControl.maxValidErrorMessage',
-        defaultMessage: 'max must be greater or equal to min'
-      });
-    }
+export function ceilWithPrecision(value, decimalPlaces) {
+  return roundWithPrecision(value, decimalPlaces, Math.ceil);
+}
 
-    this.setState({
-      minValue: min,
-      maxValue: max,
-      isRangeValid,
-      errorMessage,
-    });
+export function floorWithPrecision(value, decimalPlaces) {
+  return roundWithPrecision(value, decimalPlaces, Math.floor);
+}
 
-    if (isRangeValid && isMaxValid && isMinValid) {
-      this.handleOnChangeComplete({ min, max });
+class RangeControlUi extends Component {
+  state = {};
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const nextValue = nextProps.control.hasValue()
+      ? [nextProps.control.value.min, nextProps.control.value.max]
+      : ['', ''];
```

Comment:
I wouldn't pass blank values if the user hasn't changed them, I would set them to the min/max values available.

Reply:
The empty state is not value. Setting to min and max makes the user assume there is a filter applied which is not the case

Reply:
Doesn't the absence of values mean that the full range is in play, though?

Reply:
> Doesn't the absence of values mean that the full range is in play, though?

No, the absence of values means that there is nothing set. Logically that does mean that the entire range is in play but setting a range and having the range be unset are 2 important UI states. 
1) When the range is unset, there is not range filtering the results. The UI needs to be empty to show the user that no filtering is being applied. 
2) When a range is set, there is a range filtering the results. The UI needs to show the user the range so they can visually see how their data is being filtered. 

The min/max range will change depending on user selected time, active filters, and query. Showing the entire range as selected is misleading when no range filter is getting applied.